### PR TITLE
Add tcmalloc for amd64 platform

### PR DESCRIPTION
**Pull request by derekbit**
Depends on https://github.com/rancher/tgt/pull/8 and https://github.com/rancher/liblonghorn/pull/16

[Longhorn 4136](https://github.com/longhorn/longhorn/issues/4136)

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -20,23 +24,46 @@ COPY bin/longhorn-instance-manager /usr/local/bin/
...
+RUN if [ ${ARCH} == "amd64" ]; then \
+        cd /usr/src && \
+        git clone https://github.com/gperftools/gperftools.git&& \
+        cd gperftools && \
+        git checkout db6d94d9893fdd8c74028b5113cd9185be1e3111 && \
+        ./autogen.sh && \
+        ./configure && \
+        make && \
+        make install; \
+    fi
```
Comment (shuo-wu) [inline package/Dockerfile:40]: Is `rm -rf /usr/src/gperftools` missed here?
Similar below

**Review by shuo-wu — COMMENTED** [review]
Do we have a ticket to track the arm64 part?

File: package/Dockerfile (hunk 1 above)
Comment (derekbit) [reply package/Dockerfile:40]: Yeah, we can delete it.

**Comment by derekbit** [comment]
> Do we have a ticket to track the arm64 part?

No, need to investigate the arm64 part later.
I will create a ticket tracking arm64 part.

**Review by joshimoo — APPROVED** [review]
looks okay to me. 

Seems like a hack though relying on https://github.com/gperftools/gperftools.git installation just to get tcmalloc instead of building directly via https://github.com/google/tcmalloc but setting up bazel etc could be tedious.

Let's see if gperftools also has something for ARM64 otherwise we should go with relying/building tcmalloc directly.

File: package/Dockerfile (hunk 1 above)
Comment (derekbit) [reply package/Dockerfile:42]: Addressed. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**